driver-core/src/main/com/mongodb/internal/operation/MixedBulkWriteOperation.java

```diff
@@ -494,7 +494,7 @@ public void onResult(final BsonDocument result, final Throwable t) {
                 } else {
                     MongoException writeConcernBasedError = ProtocolHelper.createSpecialException(result,
                             connection.getDescription().getServerAddress(), "errMsg");
-                    if (writeConcernBasedError != null && shouldAttemptToRetryWrite(true, writeConcernBasedError, maxWireVersion)) {
+                    if (writeConcernBasedError != null && shouldAttemptToRetryWrite(retryWrites, writeConcernBasedError, maxWireVersion)) {
```

Comment:
The spec states that the RetryableWriteError label can only be added when `retryWrites=true`. The test case that was failing on Evergreen set `retryWrites=false`, but the call to `shouldAttemptToRewrite` would always return `true` for the exception `MongoNodeIsRecoveringException`.

Reply:
ack